src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -85,29 +86,69 @@ def calculate_timeout_seconds(
     ),
   )
 
-  test_target_sources_file_names = sorted(source_root_stripped_test_target_sources.snapshot.files)
+  return TestTargetSetup(
+    requirements_pex=resolved_requirements_pex,
+    sources_file_names=tuple(sorted(source_root_stripped_test_target_sources.snapshot.files)),
+    input_files_digest=merged_input_files
+  )
+
+
+@rule(name="Run pytest")
+async def run_python_test(
+  test_target: PythonTestsAdaptor,
+  pytest: PyTest,
+  python_setup: PythonSetup,
+  subprocess_encoding_environment: SubprocessEncodingEnvironment
+) -> TestResult:
+  """Runs pytest for one target."""
+
+  test_setup = await Get[TestTargetSetup](PythonTestsAdaptor, test_target)
```

Comment:
You can directly request this in the rule signature as a parameter `test_setup: TestTargetSetup`.